common/src/main/scala/tsec/passwordhashers/core/PasswordHashAPI.scala

```diff
@@ -0,0 +1,122 @@
+package tsec.passwordhashers.core
+
+import java.nio.CharBuffer
+
+import cats.Id
+import cats.effect.Sync
+import tsec.common._
+
+trait PasswordHashAPI[A] {
+
+  def hashpwUnsafe(p: String)(implicit P: PasswordHasher[Id, A]): PasswordHash[A] = hashpwUnsafe(p.asciiBytes)
+
+  /** Hash a password in a char array
+    * then clear the data in the password original
+    * array, as well as the byte encoding change.
+    *
+    * @param p the encoded password
+    * @return
+    */
+  def hashpwUnsafe(p: Array[Char])(implicit P: PasswordHasher[Id, A]): PasswordHash[A] = {
```

Comment:
why is this not just 
```scala
def hashpwUnsafe(p: Array[Char])(implicit P: PasswordHasher[Id, A]): PasswordHash[A] =
  P.hashpwUnsafe(p)
```
?

Reply:
Ha! It should be.

Good catch.

For background, I first wrote PasswordHashAPI _before_ I turned it into a typeclass, so that's some leftover I didn't catch.